### PR TITLE
Update docs for agent managed by fleet on kubernetes

### PR DESCRIPTION
**Pull request by MichaelKatsoulis**
Add documents for Agent installation on Kubernetes managed by Fleet.

Required kubernetes manifests in beats repo where added with https://github.com/elastic/beats/pull/26566 and updated with https://github.com/elastic/beats/pull/26756

### CODE REVIEW

**Comment by MichaelKatsoulis** [comment]
Updates added in Fleet User Guide --> Elastic Agents , files _Run Elastic Agent on Kubernetes managed by Fleet_ and 
_Run Elastic Agent standalone on Kubernetes_

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+{es}. The policy specifies how to collect observability data from the services
+to be monitored. The {agent} connects to a trusted {fleet-server} instance
+to retrieve the policy and report agent events.
+
+We recommend using {fleet} management because it makes the management and
+upgrade of agents considerably easier.
+
+On Kubernetes deploy {agent} as a https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/[DaemonSet]
+to ensure that there is a running instance on each node of the cluster.
+These instances are used to retrieve metrics from the host, such as system metrics, Docker stats,
```
Comment (ChrsMark) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc:29]:
```suggestion
These instances are used to retrieve metrics from the host, such as system metrics, container stats,
```

**Review by ChrsMark — APPROVED** [review]
It looks good overall, left a couple of suggestions.

**Comment by MichaelKatsoulis** [comment]
Hi @dedemorton ! Could you please take a look at this PR?

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+****
+====
+
+[discrete]
+=== Run {agent} on master nodes
+
+Kubernetes master nodes can use https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/[taints]
+to limit the workloads that can run on them. The manifest for managed {agent} defines
+tolerations to run on master nodes. Agents running on master nodes collect metrics from the control plane
+components (scheduler, controller manager) of Kuberentes.
```
Comment (bmorelli25) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc:111]:
```suggestion
components (scheduler, controller manager) of Kubernetes.
```

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+ tolerations:
+ - key: node-role.kubernetes.io/master
+   effect: NoSchedule
+------------------------------------------------
+
+
+[discrete]
+== Deploy
+
+If planing to deploy `state_*` datasets of Kubernetes package,
```
Comment (bmorelli25) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc:126]:
```suggestion
If planning to deploy `state_*` datasets of Kubernetes package,
```

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+------------------------------------------------
+$ kubectl get pod -n kube-system -l app=elastic-agent
+
+NAME                  READY   STATUS    RESTARTS   AGE
+elastic-agent-hrjbg   1/1     Running   0          12m
+elastic-agent-olpsd   1/1     Running   0          12m
+------------------------------------------------
+
+
+{agent}s should be enrolled to Fleet and user should be able to deploy Kubernetes package accordingly.
```
Comment (bmorelli25) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc:151]:
```suggestion
{agent}s should be enrolled to Fleet and users should be able to deploy the Kubernetes package accordingly.
```

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+However, version {version} of {agent} has not yet been
+released, so no Docker image is currently available for this version.
+
+endif::[]
+
+
+[discrete]
+== Kubernetes deploy manifests
+
+With {fleet}, each agent enrols at a policy defined in {kib} and stored in
```
Comment (bmorelli25) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-managed-by-fleet.asciidoc:19]:
```suggestion
With {fleet}, each agent enrolls in a policy defined in {kib} and stored in
```

**Review by bmorelli25 — COMMENTED** [review]
This is looking great! Just a few comments.

**Review by bmorelli25 — APPROVED** [review]
LGTM

**Comment by MichaelKatsoulis** [comment]
@bmorelli25 Is it ok to merge it ?

**Pull request merged**